libopenage/renderer/opengl/window.cpp

```diff
@@ -57,6 +57,21 @@ GlWindow::GlWindow(const std::string &title,
 	this->window->setFormat(format);
 	this->window->create();
 
+	// set display mode
+	switch (settings.mode) {
+	case window_mode::FULLSCREEN:
+		this->window->showFullScreen();
+		break;
+	case window_mode::BORDERLESS:
+		this->window->setFlags(this->window->flags() | Qt::FramelessWindowHint);
+		this->window->show();
+		break;
+	case window_mode::WINDOWED:
+	default:
+		this->window->showNormal();
+		break;
+	}
```

Comment:
You should not use the `show*` methods since they also set the window to visible (which we do not necessarily want). It's better to use the `QWindow::setWindowStates` method.